Update ietf-scitt-charter.md (#20)
ietf-scitt/charter: ietf-scitt-charter.md

Introduction
============

> A Device's supply chain, including its components, is complex often being comprised of both common libraries and custom components. Knowing the identity of a Device or its component is critical to determining its compliance. However, the scale, size, and complexity of supply chains, from design, manufacturing, logistics, and just-in-time delivery has led to more frequent and sophisticated supply chain attacks. The traditional methods of safeguarding supply chains (e.g., pre- and post-audit methodologies) are no longer adequate, therefore, with the increase in suppliers and increase in the complexity of the various supply chains the means to determine a Device or component's identity requires standardization.

Reviewer: Suggest retaining the previous wording which is more generic.

Reviewer 2: Would retain the old text for now!

1. A single product is composed of multiple sub-products coming from different suppliers. There are several standards to compose supply chain information with different producers choosing different methods.
2. There are no APIs defined for automated publishing, retrieval, or independent verification of the information above.
4. The absence of decentralized, globally interoperable, transparent services to publish the supply chain data.
5. The lack of sufficient standards for independently verifying the presence of supply chain data in tamper-proof data stores.
6. Software consumers have no trustworthy way to verify that a software signature on a software package is legitimate.

[//]: # (MW: #6 is not true, SWID tags can be signed today.)

A minimal, simple, and concise set of building blocks that interact in a standardized way will provide long-term accountability and interoperability for supply chain components throughout their lifecycles across architecturally diverse systems.

Goals
=====
Based on an input document on the architecture (draft-birkholz-scitt-architecture-00), the WG will:

> [//]: # (MW: Seems circular to have within the charter a reference to one of the documents to be produced by the WG)

Reviewer: Agree.

1. Standardize the technical flows for providing information about a software supply chain, which also includes firmware, and covering the essential building blocks that make up the architecture.
2. In addition to this, the WG shall employ the existing work already done within
    - other IETF WGs such as COSE WG, and IETF RATS WG, as appropriate,

>     - coordination with other bodies, such as the, OpenSSF, W3C, ISO, and the Trusted Computing Group.

Reviewer: LGTM

Introduction
============

> A Device's supply chain, including its components, is complex often being comprised of both common libraries and custom components. Knowing the identity of a Device or its component is critical to determining its compliance. However, the scale, size, and complexity of supply chains, from design, manufacturing, logistics, and just-in-time delivery has led to more frequent and sophisticated supply chain attacks. The traditional methods of safeguarding supply chains (e.g., pre- and post-audit methodologies) are no longer adequate, therefore, with the increase in suppliers and increase in the complexity of the various supply chains the means to determine a Device or component's identity requires standardization.

Reviewer 3: ```suggestion
Over the years, rapid technological advancements have motivated organizations to implement efficient processes for supply chains (i.e., from design to manufacturing, logistics, and just-in-time delivery).
While these improvements help organizations increase efficiency and swiftly bring innovations to market, the rapid increase in scale, size, and complexity of supply chains has led to more frequent and sophisticated supply chain attacks.
The traditional methods of safeguarding supply chains (e.g., pre- and post-audit methodologies) are no longer adequate.
```

Introduction
============

> A Device's supply chain, including its components, is complex often being comprised of both common libraries and custom components. Knowing the identity of a Device or its component is critical to determining its compliance. However, the scale, size, and complexity of supply chains, from design, manufacturing, logistics, and just-in-time delivery has led to more frequent and sophisticated supply chain attacks. The traditional methods of safeguarding supply chains (e.g., pre- and post-audit methodologies) are no longer adequate, therefore, with the increase in suppliers and increase in the complexity of the various supply chains the means to determine a Device or component's identity requires standardization.

Reviewer 4: Suggest inserting "digital" in front of device.

> [//]: # (MW: The previous introduction sounded like this is a new problem -- it isn't. It's only recently gained importance and recognition due to the increase in interconnected devices. If we are going to cite a rationale to dealing with this now it is the complex supply chain from various commonly used and complexity of component distribution and their diverse sources.)

Reviewer 5: ```suggestion
```

Reviewer 5: [//]: # (MW: The previous introduction sounded like this is a new problem -- it isn't. It's only recently gained importance and recognition due to the increase in interconnected devices. If we are going to cite a rationale to dealing with this now it is the complex supply chain from various commonly used and complexity of component distribution and their diverse sources.)

The output of the SCITT WG is a set of standards that define the essential building blocks enabling the security of supply chain systems and assisting implementers in securing deployments.

> For example, a public computer interface system could report its software composition, which can be compared against known software compositions for such a device, as recorded in a public append-only transparent registry.

Reviewer 3: ```suggestion
For example, a public computer interface system could report its software composition, which can be compared against known software compositions for such a device, as recorded in an append-only transparent registry.
```

Therefore, providing an individual using the system with confidence that it will behave as and when expected, consistently and without deviation.

> [//]: # (MW: "recorded in a public ledger, " I thought that is too implementation specific for an introduction)

Reviewer 5: agree

Reviewer: Also agree.  Additionally, I suggest a few capitalization and grammatical edits.

SCITT forms a set of building blocks that will allow implemeters to build interoperable supply chain integrity systems. For example, a public computer interface system could report its software composition that can be compared against known software compositions for such a device providing an individual using the system confidence that it will behave as expected, when expected, and nothing/nowhen else

Reviewer 3: resolved in previous commits and reflected in current branch. also removed "public" as that can be misunderstood easily

Reviewer 6: @kaywilliams I like where that is going. May I offer this slight mod. I'm concerned that we're swerving a little into the 'preventative security' world where actually the work here is transparency and accountability. In the example given we don't have total confidence that the computer will always behave perfectly: there may be bugs or operator-borne attacks, for example. But what we do have is confidence that you're dealing with the computer system you thought you would be, and that nothing has been injected or modified on its way to you. 

I also removed "an individual" because humans aren't realistically going to be doing much of this transactional verification: we want all of this stuff to be strong enough that we can automate all the mundane validation work away.

_"SCITT forms a set of interoperable building blocks that will allow implementers to build integrity and accountability into supply chain systems. For example, a public computer interface system could report its software composition that can then be compared against known software compositions (and certifications?) for such a device thereby giving confidence that the system is running the software expected and has not been modified, either by attack or accident, in the supply chain"._

> [//]: # (MW: "recorded in a public ledger, " I thought that is too implementation specific for an introduction)

Reviewer 2: Remove public from Line 8

> [//]: # (MW: Does "nowhen" belong here?)

Reviewer 5: ```suggestion
```
[//]: # (MW: Does "nowhen" belong here?)

Reviewer 3: resolved via previous commits

Problem Statement
=================
It is challenging to manage the compliance of products across end-to-end, global supply chains. 

Some of the fundamental security issues that face the supply chain ecosystem today are as follows:

1. A single product is composed of multiple sub-products coming from different suppliers. There are several standards to compose supply chain information with different producers choosing different methods.
2. There are no APIs defined for automated publishing, retrieval, or independent verification of the information above.
4. The absence of decentralized, globally interoperable, transparent services to publish the supply chain data.
5. The lack of sufficient standards for independently verifying the presence of supply chain data in tamper-proof data stores.

> 6. Software consumers have no trustworthy way to verify that a software signature on a software package is legitimate.

Reviewer 3: postponed

> [//]: # (MW: #6 is not true, SWID tags can be signed today.)

Reviewer 5: I interpret this as know standard way to discover if they key used to sign the tag is authoritative for the entity in question. SPDX and Cyclone also support signing.

> [//]: # (MW: #6 is not true, SWID tags can be signed today.)

Reviewer 5: ```suggestion

```

[//]: # (MW: #6 is not true, SWID tags can be signed today.)

While these improvements help organizations increase efficiency and swiftly bring innovations to market, the rapid increase in scale, size, and complexity of supply chains has led to more frequent and sophisticated supply chain attacks.
The traditional methods of safeguarding supply chains (e.g., pre- and post-audit methodologies) are no longer adequate.

> 

Reviewer 3: ```suggestion
```

The traditional methods of safeguarding supply chains (e.g., pre- and post-audit methodologies) are no longer adequate.

The output of the SCITT WG is a set of standards that define the essential building blocks enabling the security of supply chain systems and assisting implementers in securing deployments.
For example, a public computer interface system could report its software composition, which can be compared against known software compositions for such a device, as recorded in an append-only transparent registry.
Therefore, providing an individual using the system with confidence that it will behave as and when expected, consistently and without deviation.

> 

Reviewer 3: ```suggestion
```

1. A single product is composed of multiple sub-products coming from different suppliers. There are several standards to compose supply chain information with different producers choosing different methods.
2. There are no APIs defined for automated publishing, retrieval, or independent verification of the information above.
4. The absence of decentralized, globally interoperable, transparent services to publish the supply chain data.
5. The lack of sufficient standards for independently verifying the presence of supply chain data in tamper-proof data stores.
6. Software consumers have no trustworthy way to verify that a software signature on a software package is legitimate.

[//]: # (MW: #6 is not true, SWID tags can be signed today.)

A minimal, simple, and concise set of building blocks that interact in a standardized way will provide long-term accountability and interoperability for supply chain components throughout their lifecycles across architecturally diverse systems.

Goals
=====
Based on an input document on the architecture (draft-birkholz-scitt-architecture-00), the WG will:

> 

Reviewer 3: removing emptyline

> 

Reviewer 3: ```suggestion
```

removing empty line

1. make recommendations or suggestions on best practices on how to design the supply chain,
2. establish a universal/centralized registry for supply chain data,

> 3. define methods to prevent supply chain issuers from making false claims,

Reviewer 3: ```suggestion
3. define methods to prevent authenticated supply chain issuers from making false claims,
```

1. make recommendations or suggestions on best practices on how to design the supply chain,
2. establish a universal/centralized registry for supply chain data,
3. define methods to prevent supply chain issuers from making false claims,

> 3. try to prevent supply chain issuers from making false claims,

Reviewer 3: ```suggestion
```

1. make recommendations or suggestions on best practices on how to design the supply chain,
2. establish a universal/centralized registry for supply chain data,

> 3. define methods to prevent supply chain issuers from making false claims,

Reviewer: Agree

4. define specific implementation guidance on storage, query, or retrieval of supply chain statements, or
5. select a specific Bill of Materials (BOM) formats and metadata headers.

[//]: # (Bill of Materials \(BOM\): We don't define this term prior to use. I like including this term, shouldn't we define it as a component of the supply chain)

Program of Work
===============

The main deliverables defined by this program of work provide a guideline for milestones that are in scope of the WG charter. Documents produced by the working group will address one or more of the following main deliverables:

## Architectural Model: Actors, Interactions, Terminology

> The WG shall start by documenting and defining terms in an architectural model for:

Reviewer: Agree with this change

1. make recommendations or suggestions on best practices on how to design the supply chain,
2. establish a universal/centralized registry for supply chain data,
3. define methods to prevent supply chain issuers from making false claims,
3. try to prevent supply chain issuers from making false claims,
4. define data formats for payload content, such as Bills of Materials data formats

> [//]: # (Bill of Materials \(BOM\): We don't define this term prior to use. I like including this term, shouldn't we define it as a component of the supply chain)

Reviewer 5: ```suggestion

```

[//]: # (Bill of Materials \(BOM\): We don't define this term prior to use. I like including this term, shouldn't we define it as a component of the supply chain)

2. Notarization: The WG shall develop a specification that describes the notarization information model and the interaction model a notary will use to interact with supply chain entities.

3. Auditing: The WG shall develop standards to permit auditing of supply chain claims that are introduced in the transparent registry. This will, in turn, generate audit claims based on an information model (results of an audit), which can be introduced in the same registry. A corresponding interaction model will describe how audit information can be queried by supply chain consumers (e.g., end customers) before making critical business decisions.

## Versatile Countersigning Format in Support of Transparency Services

The WG shall specify a standard format for authenticity data returned from the transparent registry such as proofs, etc. The standard shall enable independent verification of supply chain claims at a (much) later point on multiple platforms across multiple geographical locations.

[//]: # (MW: authenticity data is not defined above. What is it?)

## Generic Protocol Bindings for Information Model and Interaction Models

> The WG shall standardize request-response interactions ("external API") and potentially other generic interaction schemes provided to various actors to interact with the supply chain ecosystem. This includes standardizing inter-component messages (based on the interaction models) between supply chain actors to support common reference implementations of SCITT building blocks by various organizations to expedite industry-wide adaptation.

Reviewer: Agree with these changes.

2. Notarization: The WG shall develop a specification that describes the notarization information model and the interaction model a notary will use to interact with supply chain entities.

3. Auditing: The WG shall develop standards to permit auditing of supply chain claims that are introduced in the transparent registry. This will, in turn, generate audit claims based on an information model (results of an audit), which can be introduced in the same registry. A corresponding interaction model will describe how audit information can be queried by supply chain consumers (e.g., end customers) before making critical business decisions.

## Versatile Countersigning Format in Support of Transparency Services

The WG shall specify a standard format for authenticity data returned from the transparent registry such as proofs, etc. The standard shall enable independent verification of supply chain claims at a (much) later point on multiple platforms across multiple geographical locations.

> [//]: # (MW: authenticity data is not defined above. What is it?)

Reviewer 5: ```suggestion

```

[//]: # (MW: authenticity data is not defined above. What is it?)

## Generic Protocol Bindings for Information Model and Interaction Models

> The WG shall standardize request-response interactions ("external API") and potentially other generic interaction schemes provided to various actors to interact with the supply chain ecosystem. This includes standardizing inter-component messages (based on the interaction models) and payload serialization between supply chain actors to support common reference implementations of SCITT building blocks by various organizations to expedite industry-wide adaptation.

Reviewer 7: this is excellent
